### PR TITLE
add createdBy and lastUpdatedBy columns

### PR DESCRIPTION
**Pull request by eperedo**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8693e3wam

### :video_camera: Screenshots/Screen capture

Columns in table

image

Columns in selector

image

### CODE REVIEW

File: src/data/repositories/UserD2ApiRepository.ts [hunk 1]
```diff
@@ -159,6 +166,18 @@ export class UserD2ApiRepository implements UserRepository {
         });
     }
 
+    private getUserByIds(userIds: Id[]) {
```
Comment (tokland) [inline src/data/repositories/UserD2ApiRepository.ts:169]: getUsersById

**Review by tokland — APPROVED** [review]
All good! (just a minor proposal of rename)

**Comment by MiquelAdell** [comment]
@eperedo just so I know, I've seen this code change:
```// in version 2.38 stats and typeReports are inside a response object
type Dhis2Response = MetadataResponse & {
    response?: { stats: MetadataResponse["stats"]; typeReports: MetadataResponse["typeReports"] };
};
```
Will that mean that the app will not be compatible with 2.37? or are we doing some kind of check to get stats from one part or the other?

I'm ok either way but I need to know what versions are we supporting once I release the app

**Comment by eperedo** [comment]
> @eperedo just so I know, I've seen this code change:
> 
> ```
> type Dhis2Response = MetadataResponse & {
>     response?: { stats: MetadataResponse["stats"]; typeReports: MetadataResponse["typeReports"] };
> };
> ```
> 
> Will that mean that the app will not be compatible with 2.37? or are we doing some kind of check to get stats from one part or the other?
> 
> I'm ok either way but I need to know what versions are we supporting once I release the app

We're checking the response so we can support both versions.

**Pull request merged**